### PR TITLE
feat(table): implement table disable state

### PR DESCRIPTION
**Pull request by epolikrati**
## Description

This PR implements the disable state of `Table` component

### CODE REVIEW

File: src/components/Table/types.ts [hunk 1]
```diff
@@ -44,6 +44,8 @@ export type TableProps = {
   onRowSelect?: (selectedRows: Row[]) => void;
   onRowClick?: (row: Row) => void;
   onHoveredRowChange?: (hoveredRow: Row | null) => void;
+  disabled?: boolean;
```
Comment (ThrasyvoulosKafasis) [inline src/components/Table/types.ts:47]: Do you really have to add these props in the props type? Is seems that the table component accept all available prop for an html table.
`export type Props = ExtendableProps<HTMLAttributes<HTMLTableElement>, TableProps>;`

**Review by ThrasyvoulosKafasis — CHANGES_REQUESTED** [review]

File: src/components/Table/types.ts (hunk 1 above)
Comment (epolikrati) [reply src/components/Table/types.ts:47]: Indeed, the className prop is part of the HTMLAttributes, fixed it here [1e2cbf7](https://github.com/epignosis/gnosis/pull/303/commits/1e2cbf7b38fb556485a0b74a27960e89ab8e4cca). However, the disabled attribute is not applicable to <table> elements, so we will need to include it in our types.

**Review by lkatmadas — APPROVED** [review]

File: src/components/Table/components/Body.tsx [hunk 2]
```diff
@@ -12,6 +12,7 @@ const Body: FC<ChildrenProps> = ({
   dispatch,
   onRowClick,
   onHoveredRowChange,
+  disabled,
```
Comment (lkatmadas) [inline src/components/Table/components/Body.tsx:15]: Should we set these disabled props to false by default? (the same applies in other files too)

Comment (epolikrati) [reply src/components/Table/components/Body.tsx]: You're right. Setting the disabled props to false by default is a good practice :slightly_smiling_face: 
[4617f34](https://github.com/epignosis/gnosis/pull/303/commits/4617f34755f465b7c5d54225b1526aa91674334d)

**Review by lkatmadas — APPROVED** [review]

**Review by ThrasyvoulosKafasis — APPROVED** [review]

**Pull request merged**